### PR TITLE
add support for ymd75 rev3

### PR DESCRIPTION
**Pull request by SoMuchForSubtlety**
This commit adds support for [rev3 of ymd75](https://www.aliexpress.com/item/1005001370568297.html).

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by ridingqwerty — APPROVED** [review]

File: keyboards/ymd75/rev3/rules.mk [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = atmel-dfu
+
+# Build Options
+BOOTMAGIC_ENABLE = yes      # Virtual DIP switch configuration
```
Comment (fauxpark) [inline keyboards/ymd75/rev3/rules.mk:8]:
```suggestion
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = full     # Virtual DIP switch configuration
```

File: keyboards/ymd75/rev3/rules.mk [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+
+KEY_LOCK_ENABLE = yes	    # Enable KC_LOCK support
```
Comment (fauxpark) [inline keyboards/ymd75/rev3/rules.mk:25]:
```suggestion
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
KEY_LOCK_ENABLE = yes       # Enable KC_LOCK support
```

**Review by fauxpark — APPROVED** [review]
Might as well remove `DESCRIPTION` too while we're at it :)

**Review by fauxpark — APPROVED** [review]

File: keyboards/ymd75/config.h [hunk 3]
```diff
@@ -23,15 +23,13 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
...
-#define DEVICE_VER      0x0200
 
 #define MANUFACTURER    YMDK
 #define PRODUCT         YMD75 / MT84
-#define DESCRIPTION     75% Keyboard
-
 
 #define BACKLIGHT_PIN D4
 #define BACKLIGHT_LEVELS 12
+#define RGBLIGHT_EFFECT_KNIGHT_OFFSET 4
```
Comment (zvecr) [inline keyboards/ymd75/config.h:32]: Is this applicable for all revisions, or just the newly added rev3?

Comment (SoMuchForSubtlety) [reply keyboards/ymd75/config.h:32]: I have not tested it on previous versions. I can move it to the rev3 config.

**Pull request merged**